src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -0,0 +1,806 @@
+                                                        S,
+                                            precision_of<RT>::value);
+                        m_attn_w.at({b, h, pq, pk}) = sum;
+                    }
+                }
+            });
+        } else {
+            parallel_for3d(B, H, kv_len, [&](size_t b, size_t h, size_t pk) {
+                // which batch item should be used at postion pk?
+                auto b_kv = beams ? beams.at({b, pk}) : b;
+                for (size_t pq = 0; pq < q_len; pq++) {
+                    auto sum = attn_dot_product(&query.at({b, h, pq, 0}),
+                                                      &present_key.at({b_kv, h, pk, 0}),
+                                                      S,
+                                                      precision_of<RT>::value);
+                    m_attn_w.at({b, h, pq, pk}) = sum;
+                }
+            });
+        }
+
+        parallel_for3d(B, H, q_len, [&](size_t b, size_t h, size_t pq) {
+            // apply attention mask & sofmax
+            auto ncausal = auto_causal ? (kv_len - q_len + pq + 1) : kv_len;
+            float* alibi_ptr = alibi_mask ? &alibi_mask.at({b, h, pq, 0}, true) : nullptr;
+            float* attn_mask_ptr = attention_mask ? &attention_mask.at({b, h, pq, 0}, true) : nullptr;
+            uint8_t* cmask_ptr = causal_mask ? &causal_mask.at({b, h, pq, 0}, true) : nullptr;
+            attn_softmax(&m_attn_w.at({b, h, pq, 0}),
+                         &m_attn_w.at({b, h, pq, 0}),
+                         d_scale,
+                         alibi_ptr,
+                         attn_mask_ptr,
+                         cmask_ptr,
+                         select_nfltmax_at_0,
+                         ncausal,
+                         kv_len,
+                         Precision::FP32);
+        });
+
+        // attn_w * V
+        auto nthr = parallel_get_max_threads();
+        m_temp.resize({static_cast<size_t>(nthr), B, q_len, H, S});
+        // m_attn_w {B, H, q_len, kv_len}
+        if (h_each_group_len == 0) {
+            parallel_nt_static(nthr, [&](const size_t ithr, const size_t nthr) {
+                size_t start{0}, end{0};
+                splitter(B * H * kv_len, nthr, ithr, start, end);
+
+                memset(&m_temp.at({ithr, 0, 0, 0, 0}), 0, m_temp.stride(0) * sizeof(float));
+
+                size_t b, h, pv;
+                parallel_it_init(start, b, B, h, H, pv, kv_len);
+                for (size_t iwork = start; iwork < end; ++iwork) {
+                    auto b_kv = beams ? beams.at({b, pv}) : b;
+                    auto* v = &present_value.at({b_kv, h, pv, 0});
+                    for (size_t pq = 0; pq < q_len; pq++) {
+                        auto* out = &m_temp.at({ithr, b, pq, h, 0});
+                        auto weight = m_attn_w.at({b, h, pq, pv});
+                        attn_acc_value(out, weight, v, S, precision_of<RT>::value);
```

Comment:
the same concern here, about possible dispatch overhead

Reply:
Moved the loop into `attn_acc_value`.